composer/core/tracing.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+from typing import Any, Optional, Tuple
+
+import torch
+from torchmetrics.collections import MetricCollection
+
+from composer.core.types import Batch, Metrics, Tensors
+from composer.trainer.trainer import BaseMosaicModel
+from composer.utils.device_helpers import move_batch_to_device
+
+
+def trace_mosaic_model(model: BaseMosaicModel,
+                       example_input: Batch,
+                       device: str = "cpu",
+                       save_file_path: Optional[str] = None):
+    if not device in ("cpu", "gpu"):
+        raise ValueError(f"Invalid device {device} must be either 'cpu' or 'gpu'.")
+
+    if device == "gpu":
+        # just use one GPU for tracing
+        torch_device = torch.device(f"cuda:0")
+        model.to(torch_device)
+        example_input = move_batch_to_device(batch=example_input, device=torch_device)
+
+    output = model.forward(batch=example_input)
+
+    jit_model = torch.jit.trace_module(model, {'forward': (example_input,), 'loss': (output, example_input)})
+
+    if save_file_path:
+        torch.jit.save(jit_model, save_file_path)
+
+    return jit_model
+
+
+def load_model_trace(filename: str) -> BaseMosaicModel:
+    jit_model = torch.jit.load(filename)
+
+    model = _JITTracedMosaicModel()
+    model.forward = jit_model.forward  # type: ignore
```

Comment:
I missed this before, would prefer something like `model = JITTracedMosaicModel(jit_model)` rather than monkeypatching.

Reply:
Then no need for the type ignores